pkg/ipamd/ipamd.go

```diff
@@ -783,7 +835,36 @@ func (c *IPAMContext) tryAssignIPs() (increasedPool bool, err error) {
 			ipamdErrInc("increaseIPPoolGetENIaddressesFailed")
 			return true, errors.Wrap(err, "failed to get ENI IP addresses during IP allocation")
 		}
-		c.addENIaddressesToDataStore(ec2Addrs, eni.ID)
+
+		c.addENIsecondaryIPsToDataStore(ec2Addrs, eni.ID)
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c *IPAMContext) tryAssignPrefixes() (increasedPool bool, err error) {
+	toAllocate := c.getPrefixesNeeded()
+	// Returns an ENI which has space for more prefixes to be attached, but this
+	// ENI might not suffice the WARM_IP_TARGET/WARM_PREFIX_TARGET
+	eni := c.dataStore.GetENINeedsIP(c.maxPrefixesPerENI, c.useCustomNetworking)
+	if eni != nil {
+		currentNumberOfAllocatedPrefixes := len(eni.AvailableIPv4Cidrs)
+		err = c.awsClient.AllocIPAddresses(eni.ID, min((c.maxPrefixesPerENI-currentNumberOfAllocatedPrefixes), toAllocate))
```

Comment:
nit: we should hide two different API calls behind a single AllocIPAddresses

Reply:
Agreed, will fix it up post GA.